### PR TITLE
OADP-4291 oadp-1.4 Add no-pod/immediate bind VM tests.

### PR DESCRIPTION
**Pull request by mrnold**
## Why the changes were made

For [OADP-4291](https://issues.redhat.com//browse/OADP-4291), adds requested tests for no-pod VM backup/restore and immediate volume binding mode. Not yet merged to master, waiting for openshift/velero konveyor-dev rebase.

## How to test the changes made

make TEST_VIRT=true test-e2e

### CODE REVIEW

File: tests/e2e/lib/virt_helpers.go [hunk 1]
```diff
@@ -185,14 +185,14 @@ func (v *VirtOperator) checkNamespace() bool {
 // Checks for the existence of the virtualization operator group
 func (v *VirtOperator) checkOperatorGroup() bool {
 	group := operatorsv1.OperatorGroup{}
-	err := v.Client.Get(context.TODO(), client.ObjectKey{Namespace: v.Namespace, Name: "kubevirt-hyperconverged-group"}, &group)
+	err := v.Client.Get(context.Background(), client.ObjectKey{Namespace: v.Namespace, Name: "kubevirt-hyperconverged-group"}, &group)
```
[the remark this answers is not in the document]
Comment (mpryc) [reply tests/e2e/lib/virt_helpers.go:188]: Ok for what I read the Background is for the context which doesn't really need to be cancelled or timeout'ed, that's why I'm asking.

File: tests/e2e/lib/virt_storage_helpers.go [hunk 2]
```diff
@@ -203,3 +206,41 @@ func (v *VirtOperator) CreateDataSourceFromPvc(namespace, name string) error {
...
+	}
+	if defaultStorageClass == nil {
+		return errors.New("no default storage class found")
+	}
+
+	immediateStorageClass := defaultStorageClass
+	immediateStorageClass.VolumeBindingMode = ptr.To[storagev1.VolumeBindingMode](storagev1.VolumeBindingImmediate)
+	immediateStorageClass.Name = name
+	immediateStorageClass.ResourceVersion = ""
+	immediateStorageClass.Annotations["storageclass.kubernetes.io/is-default-class"] = "false"
```
[the remark this answers is not in the document]
Comment (mrnold) [reply tests/e2e/lib/virt_storage_helpers.go]: Sure, done.

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by mpryc — APPROVED** [review]
/lgtm

File: tests/e2e/virt_backup_restore_suite_test.go [hunk 3]
```diff
@@ -223,6 +231,73 @@ var _ = Describe("VM backup and restore tests", Ordered, func() {
...
+			PowerState: "Stopped",
+			BackupRestoreCase: BackupRestoreCase{
+				Namespace:         "cirros-test",
+				Name:              "cirros-test",
+				SkipVerifyLogs:    true,
+				BackupRestoreType: lib.CSIDataMover,
+				BackupTimeout:     20 * time.Minute,
+				PreBackupVerify:   vmPoweredOff("cirros-test", "cirros-test"),
+			},
+			RestoreErr: errors.New("error to expose snapshot: error to wait target PVC consumed"),
```
Comment (mateusoliveira43) [inline tests/e2e/virt_backup_restore_suite_test.go:246]: why are we adding a test that is expected to fail in 1.4 branch?

Comment (mrnold) [reply tests/e2e/virt_backup_restore_suite_test.go:246]: I probably should have marked this "do not merge", or taken this one test out instead. But it looks like it's the only one consistently failing on 1.4, I am still looking into why.

**Pull request merged**